spec/account/account-manager.spec.js

```diff
@@ -0,0 +1,149 @@
+
+        AccountManager = require("../../lib/account/account-manager");
+        otDateMock = require("../mock/ot-date-mock");
+        promiseMock = require("../mock/promise-mock");
+        randomMock = require("../mock/random-mock");
+        accountServiceFake = jasmine.createSpyObj("accountServiceFake", [
+            "completeAsync",
+            "getAsync",
+            "initiateAsync"
+        ]);
+        accountServiceFake.completeAsync = jasmine.createSpy("complete").andCallFake(() => {
+            return promiseMock.resolve(true);
+        });
+        accountServiceFake.getAsync = jasmine.createSpy("get").andCallFake(() => {
+            return promiseMock.resolve({
+                mfaKey: "339r93939303093"
+            });
+        });
+        accountServiceFake.initiateAsync = jasmine.createSpy("initiate").andCallFake((accountId, accountInfo, options) => {
+            return promiseMock.resolve({
+                accountId: accountInfo.accountId,
+                email: accountInfo.email,
+                mfaKey: accountInfo.mfaKey,
+                salt: accountInfo.salt
+            });
+        });
+        hotpFake = jasmine.createSpyObj("hotpFake", [
+            "generateSecretAsync",
+            "verifyToken"
+        ]);
+        hotpFake.generateSecretAsync = jasmine.createSpy("generateSecretAsync").andCallFake(() => {
+            return promiseMock.resolve("thisisasecrectcodefrommfa");
+        });
+        hotpFake.verifyToken = jasmine.createSpy("verifyToken").andCallFake((key, token, opts) => {
+            if (token === "987654") {
+                return false;
+            }
+
+            return true;
+        });
+        config = {
+            account: {
+                completeLifetime: {
+                    months: 6
+                },
+                initiateLifetime: {
+                    hours: 1
+                }
+            },
+            hotp: {
+                previous: {
+                    afterDrift: 1,
+                    beforeDrift: 1
+                }
+            }
+        };
+        accountManager = new AccountManager(accountServiceFake, config, hotpFake, otDateMock, randomMock, promiseMock);
+    });
+    describe(".initiate()", () => {
+        it("gets an object back from service", (done) => {
+            accountManager.initiateAsync({
+                email: "some.one@example.net"
+            }).then((result) => {
+                expect(result).toEqual({
+                    accountId: jasmine.any(Buffer),
+                    email: "some.one@example.net",
+                    mfaKey: "thisisasecrectcodefrommfa",
+                    salt: jasmine.any(Buffer)
+                });
+                expect(result.accountId.length).toBe(24);
+                expect(result.salt.length).toBe(128);
+
+            }).then(done, done);
+        });
+    });
+    describe(".completeAsync()", () => {
+        var expectError, fail;
+
+        // TODO: Create custom matchers
```

Comment:
Yeah, that's part of a future feature that was brought up as part of the pull request, but we aren't doing it now.